pkg/util/cluster/client_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2023 The KubeAdmiral Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/kubewharf/kubeadmiral/pkg/controllers/common"
+)
+
+func TestPopulateAuthDetailsFromSecret(t *testing.T) {
+	tests := []struct {
+		name              string
+		clusterConfig     *rest.Config
+		insecure          bool
+		inputSecret       *corev1.Secret
+		useServiceAccount bool
+		wantTokenKind     string
+		wantErr           bool
+		expectErr         error
+	}{
+		{
+			name:              "normal test, insecure=true",
```

Comment:
On the structuring of unit tests: I prefer to fully specify the desired output instead of conditionally checking the desired fields (by partially duplicating the logic of the function being tested). This makes the test cases more communicative.

Reply:
The code has been updated.